README.md
# PHP guestbook

Lazarus Guestbook

By Stewart Souter

URL: http://carbonize.co.uk

Advanced Guestbook 2.3.x (PHP/MySQL)

Copyright (c)2001 Chi Kien Uong

URL: http://www.proxy2.de

Requirements:

  - MySQL 3.22.x or higher
  - PHP 5.2.x or higher

#Installation:

1. Open the configuration file 'config.inc.php' with a text editor
   and set up your MySQL database settings. 

2. Upload the guestbook to your website.

3. Run the install script http://www.yourDomain.com/install.php or http://localhost/<project_folder_name>/install.php
   and follow the instructions.

4. Give write permissions to these directories:

    - public - 777 (drwxrwxrwx) (directory)
    - tmp    - 777 (drwxrwxrwx) (directory)
    
5. The default account is:

   Username : test
   Password : 123

   But you can specify different ones in the install process.

6. Delete the install.php file.

#Updating:

1. Make a backup of your current guestbook files and if you know how the database.

2. Upload new files to your guestbook folder replacing the current ones.

3. Visit install.php in your web browser and choose update.

4. Delete the install.php file.


changelog.txt
Changelog for 1.25

Fixed - Should work with PHP 7 now
Fixed - Textarea content no longer censored when previewing
Fixed - Database connection errors now correctly explained
Fixed - Installer will display database connection errors
Fixed - Installer was hard coded to use mysql file
Fixed - captcha.php was hardcoded to use mysql.class.php
Fixed - Smileys not shown in admin entries if disabled

------------------------------------------------

Changelog for 1.24

Fixed - Comment now uses anti bot test text if question and answer is used
Fixed - Cookie login not working for some users
Fixed - Some users were getting error of Database has gone away

------------------------------------------------

Changelog for 1.23

Fixed - SFS check not returning error on comments
Fixed - Admin area not using chosen database functions

Added - More domains to bad list
Added - String escaping when saving changes in admin for security and bug fixing
Added - Ability to select all entries on admin page in one click

Removed - asianet.co.th from bad hosts list as it was blocking legitimate users

Some stuff I've forgotten 

------------------------------------------------

Changelog for 1.22

Fixed - User pics not being deleted with post
Fixed - Bad word checking was case sensitive
Fixed - Posts from Chrome on Android were showing as Chrome not Android
Fixed - Updated YouTube AGCode to support https urls

Changed - Space for name, email and URL increased to 150 to allow for non-western characters
Changed - Now just uses remote_addr for the posters IP for security and consistency

Added - Location now has a globe image that links to Google map of the location
Added - Support for iPad in browser detection
Added - Support for URLs that use non-Latin characters 

------------------------------------------------

Changelog for 1.21

Fixed - Logout in admin was not deleting your cookie
Fixed - Seeing \n or \r\n in email notifications.
Fixed - Regression meaning thumbnails were displayed wrong size
Fixed - Image not showing for Opera browser

Changed - .htaccess now has more entries for better blocking on supporting servers
Changed - StopForumSpam address now uses their load balancer

------------------------------------------------

Changelog for 1.20.4

Fixed - MySQL error if ICQ disabled or not submitted
Fixed - Entries not showing. I think browser detect was the issue.

------------------------------------------------

Changelog for 1.20.3

Fixed - New Opera detection resulted in preg_match error

------------------------------------------------

Changelog for 1.20.2

Fixed - SFS not working when both email and IP being checked and using fallback method
Fixed - Error on saving security settings if ' used in bot test question or answer
Fixed - Error message on some servers about undefined constant IS_INSTALLER
Fixed - Image Upload setting would always say directories were unwriteable
Fixed - Uploaded images not showing in easy admin
Fixed - Flash AGcode not working with youtu.be urls
Fixed - Slashes being needlessly stripped from email body
Fixed - mysql.class.php always being used even if mysqli requested

Changed - Faster deletion of redundant files in tmp folder

Added - Detection for Opera 15 and IE 11 useragents due to massive changes in useragents
Added - lazTop ID to top div so you can style the sign and navigation links

------------------------------------------------

Changelog for 1.20.1

Fixed - Entry being blocked when SFS check failed
Fixed - SFS check unable to connect on some server
Fixed - 'mysqli_insert_id() expects parameter 1 to be mysqli' error

Changed - Smileybox now has a maximum height of 150px. Scrollbar will show if needed

Removed - Some redundant code that served no purpose so god knows why I even put it in there

------------------------------------------------

Changelog for 1.20

Changed - Returned to mysql_* functions as default with option to use mysqli_* 

Added - IP addresses blocked by SFS are now cached locally for 24 hours to ease load on SFS
Added - More server IP blocks to the htaccess
Added - Header checks now checks posters hostname against list of bad hosts/server
Added - If SFS check fails then entry is marked for moderation (unaccepted)
Added - If honeypot filled in script will pause to slow down some bots

------------------------------------------------

Changelog for 1.19.1

Fixed - Regression that caused homepage url not to be saved

------------------------------------------------

Changelog for 1.19

Fixed - split function deprecated as of PHP 5.3 now using explode
Fixed - mysql functions deprecated as of PHP 5.5 switched to mysqli
Fixed - Links in RSS feed point to wrong location if using include file
Fixed - Honeypot was not working
Fixed - Timezone not set error in blocks.php

Added - Can now specify a banned IP range as x.x.x.x-y.y.y.y
Added - index, addentry and comment pages will forward to included page or block if set to
Added - Stop Forum Spam check now also checks the email address
Added - Donate link to admin menu
Added - More spam server ip blocks to .htaccess

Changed - Smiley box now slides in to view above the message textarea
Changed - Comments now slide in to view if set to hidden
Changed - Image directory names are now hardcoded due to some CMS wiping variables
Changed - Updated the .htaccess to add new IP ranges

------------------------------------------------

Changelog for 1.18.7

Fixed - Error message when saving security section of admin
Fixed - Label issues in panel_main.php

------------------------------------------------

Changelog for 1.18.6

Changed - Spam block stats now gives a detailed breakdown
Changed - Captcha no longer does a header check if enabled
Changed - Admin menu is no longer fixed to make using it on a phone easier
Changed - Captcha URL is now obfuscated in the HTML

Added - External file for displaying the block count on your website

------------------------------------------------

Changelog for 1.18


index.php
<?php 
/*
 * ----------------------------------------------
 * Lazarus Guestbook
 * by Stewart Souter
 * URL: www.carbonize.co.uk 
 * Based on Advanced Guestbook 2.3.x (PHP/MySQL)
 * Copyright (c)2001 Chi Kien Uong
 * URL: http://www.proxy2.de
 * Last Modified: Tue, 30 July 2013 20:35:24 GMT
 * ----------------------------------------------
 */

//LAZ_INCLUDE_PATH = dirname(__FILE__);
define('LAZ_INCLUDE_PATH', dirname(__FILE__));
global $GB_DB, $GB_PG;
include_once LAZ_INCLUDE_PATH.'/admin/version.php';
include_once LAZ_INCLUDE_PATH.'/admin/config.inc.php';
include_once LAZ_INCLUDE_PATH.'/lib/' . $DB_CLASS;
include_once LAZ_INCLUDE_PATH.'/lib/image.class.php';
include_once LAZ_INCLUDE_PATH.'/lib/template.class.php';

define('LAZ_TABLE_PREFIX', $table_prefix);

define('IS_INCLUDE', false);

//
// Is the guestbook being used as a CMS module?
//

if (IS_MODULE)
{
   if (!preg_match('/modules.php/i', $_SERVER['PHP_SELF']))
   {
      die ("You can't access this file directly...");
   }
   $ModName = basename(dirname( __FILE__ ));

   ob_start();
   include('header.php');

   $agbook = (isset($_GET['agbook'])) ? $_GET['agbook'] : '';
   
   function display_entries()
   {
      global $GB_PG, $ModName;
      include_once LAZ_INCLUDE_PATH.'/lib/gb.class.php';
      $gb = new guestbook(LAZ_INCLUDE_PATH);
      $GB_PG['base_url'] = $gb->db->VARS['base_url'];
      $GB_PG['index']    = $gb->db->VARS['laz_url'].'&amp;op=modload&amp;file=index';
      $GB_PG['admin']    = $gb->db->VARS['base_url'].'/admin.php';
      $GB_PG['comment']  = $gb->db->VARS['laz_url'].'&amp;op=modload&amp;file=index&amp;agbook=comment';
      $GB_PG['addentry'] = $gb->db->VARS['laz_url'].'&amp;op=modload&amp;file=index&amp;agbook=addentry';      
      $entry = (isset($_GET['entry'])) ? $_GET['entry'] : 0;
      $entry = (isset($_POST['entry'])) ? $_POST['entry'] : $entry;
      $entry = (intval($entry) < 0) ? 0 : intval($entry);
      $gb->searchfield = (isset($_GET['searchfield'])) ? trim($_GET['searchfield']) : '';
      $gb->searchtext = (isset($_GET['searchtext'])) ? trim(urldecode($_GET['searchtext'])) : '';
      $gb->searchfield = (isset($_POST['searchfield'])) ? trim($_POST['searchfield']) : $gb->searchfield;
      $gb->searchtext = (isset($_POST['searchtext'])) ? trim(urldecode($_POST['searchtext'])) : $gb->searchtext;         
      echo $gb->show_entries($entry);
      $gb->db->close_db();
   }   

   switch ($agbook)
   {

      case 'comment':
         include_once LAZ_INCLUDE_PATH.'/lib/vars.class.php';
         include_once LAZ_INCLUDE_PATH.'/lib/comment.class.php';
         $gb_com = new gb_comment(LAZ_INCLUDE_PATH);
         if ($gb_com->db->VARS['disablecomments'] > 0)
         {
            display_entries();
            break;
         } 
         $GB_PG['base_url'] = $gb_com->db->VARS['base_url'];
         $GB_PG['index']    = $gb_com->db->VARS['laz_url'].'&amp;op=modload&amp;file=index';
         $GB_PG['admin']    = $gb_com->db->VARS['base_url'].'/admin.php';
         $GB_PG['comment']  = $gb_com->db->VARS['laz_url'].'&amp;op=modload&amp;file=index&amp;agbook=comment';
         $GB_PG['addentry'] = $gb_com->db->VARS['laz_url'].'&amp;op=modload&amp;file=index&amp;agbook=addentry';
         $antispam = $gb_com->db->VARS['antispam_word'];
         $gb_com->id = (isset($_GET['gb_id'])) ? $_GET['gb_id'] : '';
         $gb_com->id = (isset($_POST['gb_id'])) ? $_POST['gb_id'] : $gb_com->id;
         $gb_com->id = intval($gb_com->id);
         $gb_com->comment = (isset($_POST['gb_comment'])) ? $_POST['gb_comment'] : '';
         $gb_com->timehash = (isset($_POST['gb_timehash'])) ? $_POST['gb_timehash'] : '';
         if(($gb_com->db->VARS['solve_media'] == 1) && ($gb_com->db->VARS['antibottest'] == 2))
         {
           $gb_com->bottest = (isset($_POST['adcopy_response'])) ? trim($_POST['adcopy_response']) : '';
         }
         else
         {
           $gb_com->bottest = (isset($_POST['gb_bottest'])) ? trim($_POST['gb_bottest']) : '';
         }
         $gb_com->user = (isset($_POST['gb_user'])) ? $_POST['gb_user'] : '';
         $gb_com->email = (isset($_POST['gb_email'])) ? $_POST['gb_email'] : '';
         $gb_com->honeypot = (isset($_POST['gb_username'])) ?  1 : 0;
         $gb_com->pass_comment = (isset($_POST['pass_comment'])) ? trim($_POST['pass_comment']) : '';
         $gb_action = (isset($_POST['gb_action'.$antispam])) ? $_POST['gb_action'.$antispam] : '';
         $gb_com->comment_action($gb_action);
         $gb_com->db->close_db();
         break;

      case 'addentry':
         include_once LAZ_INCLUDE_PATH.'/lib/vars.class.php';
         include_once LAZ_INCLUDE_PATH.'/lib/add.class.php';
         $gb_post = new addentry(LAZ_INCLUDE_PATH);
         $GB_PG['base_url'] = $gb_post->db->VARS['base_url'];
         $GB_PG['index']    = $gb_post->db->VARS['laz_url'].'&amp;op=modload&amp;file=index';
         $GB_PG['admin']    = $gb_post->db->VARS['base_url'].'/admin.php';
         $GB_PG['comment']  = $gb_post->db->VARS['laz_url'].'&amp;op=modload&amp;file=index&amp;agbook=comment';
         $GB_PG['addentry'] = $gb_post->db->VARS['laz_url'].'&amp;op=modload&amp;file=index&amp;agbook=addentry';
         $antispam = $gb_post->db->VARS['antispam_word'];
         if (isset($_POST['agb_submit_'.$antispam]) || isset($_POST['agb_preview_'.$antispam]))
         {
            $gb_post->name = (isset($_POST['gb_name'])) ? $_POST['gb_name'] : '';
            $gb_post->email = (isset($_POST['gb_email'])) ? $_POST['gb_email'] : '';
            $gb_post->url = (isset($_POST['gb_url'])) ? $_POST['gb_url'] : '';
            $gb_post->comment = (isset($_POST['gb_comment'])) ? $_POST['gb_comment'] : '';
            $gb_post->location = (isset($_POST['gb_location'])) ? $_POST['gb_location'] : '';
            $gb_post->icq = (isset($_POST['gb_icq'])) ? $_POST['gb_icq'] : 0;
            $gb_post->aim = (isset($_POST['gb_aim'])) ? $_POST['gb_aim'] : '';
            $gb_post->msn = (isset($_POST['gb_msn'])) ? $_POST['gb_msn'] : '';
            $gb_post->yahoo = (isset($_POST['gb_yahoo'])) ? $_POST['gb_yahoo'] : '';
            $gb_post->skype = (isset($_POST['gb_skype'])) ? $_POST['gb_skype'] : '';
            $gb_post->timehash = (isset($_POST['gb_timehash'])) ? $_POST['gb_timehash'] : '';
            if(($gb_post->db->VARS['solve_media'] == 1) && ($gb_post->db->VARS['antibottest'] == 2))
            {
              $gb_post->bottest = (isset($_POST['adcopy_response'])) ? trim($_POST['adcopy_response']) : '';
            }
            else
            {
              $gb_post->bottest = (isset($_POST['gb_bottest'])) ? trim($_POST['gb_bottest']) : '';
            }
            $gb_post->gender = (isset($_POST['gb_gender'])) ? $_POST['gb_gender'] : '';
            $gb_post->keep_pic = (isset($_POST['keep_pic'])) ? 1 : 0;
            $gb_post->userfile = (isset($_FILES['userfile']['tmp_name']) && $_FILES['userfile']['tmp_name'] != "") ? $_FILES : '';
            $gb_post->user_img = ((isset($_POST['gb_user_img'])) && ($gb_post->keep_pic == 1) && (empty($gb_post->userfile))) ? $_POST['gb_user_img'] : '';
            $gb_post->preview = (isset($_POST['gb_preview'])) ? 1 : 0;
            $gb_post->private = (isset($_POST['gb_private'])) ? 1 : 0;
            $gb_post->honeypot = (isset($_POST['gb_username'])) ?  1 : 0;
            $gb_action = (isset($_POST['agb_preview_'.$antispam])) ? 'preview' : 'submit';
            echo $gb_post->process($gb_action);
         }
         else
         {
            echo $gb_post->process();
         }
         $gb_post->db->close_db();
         break;

      default:
         include_once LAZ_INCLUDE_PATH.'/lib/vars.class.php';
         display_entries();
   }
   ob_end_flush();
   $base_path = dirname(dirname(LAZ_INCLUDE_PATH));
   //chdir("$base_path");
   include($base_path.'/footer.php');

}
else
{
      include_once LAZ_INCLUDE_PATH.'/lib/vars.class.php';
      include_once LAZ_INCLUDE_PATH.'/lib/gb.class.php';
      $gb = new guestbook(LAZ_INCLUDE_PATH);
      if($gb->db->VARS['included'] == 1)
      {
        header("HTTP/1.1 301 Moved Permanently");
        header("Location: " . $gb->db->VARS['laz_url']);
        exit;
      }
      elseif($gb->db->VARS['included'] == 2)
      {
        header("HTTP/1.0 404 Not Found");
        exit;
      }
      $GB_PG['base_url'] = $gb->db->VARS['base_url'];
      $GB_PG['index']    = $gb->db->VARS['base_url'].'/index.php';
      $GB_PG['admin']    = $gb->db->VARS['base_url'].'/admin.php';
      $GB_PG['comment']  = $gb->db->VARS['base_url'].'/comment.php';
      $GB_PG['addentry'] = $gb->db->VARS['base_url'].'/addentry.php';
      $entry = (isset($_GET['entry'])) ? $_GET['entry'] : 0;
      $entry = (isset($_POST['entry'])) ? $_POST['entry'] : $entry;
      $entry = intval($entry);
      $gb->searchfield = (isset($_GET['searchfield'])) ? trim($_GET['searchfield']) : '';
      $gb->searchtext = (isset($_GET['searchtext'])) ? trim(urldecode($_GET['searchtext'])) : '';
      $gb->searchfield = (isset($_POST['searchfield'])) ? trim($_POST['searchfield']) : $gb->searchfield;
      $gb->searchtext = (isset($_POST['searchtext'])) ? trim(urldecode($_POST['searchtext'])) : $gb->searchtext;     
      echo $gb->show_entries($entry);
}

?>